core/src/main/scala/chisel3/internal/package.scala

```diff
@@ -143,6 +144,14 @@ package object internal {
     }
   }
 
+  // This is similar to cats.Eval.later but we don't depend on Cats
+  private[chisel3] class Delayed[A](a: => A) {
```

Comment:
i know this is private except within `chisel3` but it's also public for developers of this codebase (?) could we add some scaladoc for it

Reply:
Will do